lua/framework/math/functions/lua_spatial_material_function.cc

```diff
@@ -91,14 +96,13 @@ LuaResponseFunction::Evaluate(int num_groups, const opensn::Vector3& xyz, int ma
   lua_pop(L, 1); // pop the table, or error code
 
   // Check return value
-  ChiLogicalErrorIf(lua_return.size() > response.size(),
-                    "Call lua-function, " + lua_function_name_ + ", returned a vector of size " +
-                      std::to_string(response.size()) +
-                      " which is greater than the number of groups " + std::to_string(num_groups) +
-                      ".");
-
-  for (size_t g = 0; g < num_groups; ++g)
-    response[g] = lua_return[g];
+  ChiLogicalErrorIf(lua_return.size() != response.size(),
+                    "Call to lua function " + lua_function_name_ + " returned a vector of " +
+                      "size " + std::to_string(lua_return.size()) + ", which is not the same as " +
+                      "the number of groups " + std::to_string(num_components) + ".");
+
+  for (size_t c = 0; c < num_components; ++c)
```

Comment:
I would advocate for std::copy here. In fact, if I read this routine correctly, I would just return `lua_return` and avoid the copy all together. You can allocate and return a vector of 1.0s above if the function name is empty.

Reply:
Good point. This was here before. Changing it.